mongo/integration/unified/entity.go

```diff
@@ -401,6 +429,178 @@ func (em *EntityMap) addDatabaseEntity(entityOptions *entityOptions) error {
+
+		if awsSessionToken != nil {
+			defaultAccessKeyID = envAwsTempAccessKeyID
+			defaultSecretAccessKey = envAwsTempSecretAccessKey
+		}
+
+		awsAccessKeyID, err := getKmsCredential(aws, "accessKeyId", defaultAccessKeyID)
+		if err != nil {
+			return err
+		}
+		if awsAccessKeyID != nil {
+			kmsProviders["aws"]["accessKeyId"] = *awsAccessKeyID
+		}
+
+		awsSecretAccessKey, err := getKmsCredential(aws, "secretAccessKey", defaultSecretAccessKey)
+		if err != nil {
+			return err
+		}
+		if awsSecretAccessKey != nil {
+			kmsProviders["aws"]["secretAccessKey"] = *awsSecretAccessKey
+		}
+	}
+
+	if azure, ok := ceo.KmsProviders["azure"]; ok {
+		kmsProviders["azure"] = make(map[string]interface{})
+
+		azureTenantID, err := getKmsCredential(azure, "tenantId", envAzureTenantID)
+		if err != nil {
+			return err
+		}
+		if azureTenantID != nil {
+			kmsProviders["azure"]["tenantId"] = *azureTenantID
+		}
+
+		azureClientID, err := getKmsCredential(azure, "clientId", envAzureClientID)
+		if err != nil {
+			return err
+		}
+		if azureClientID != nil {
+			kmsProviders["azure"]["clientId"] = *azureClientID
+		}
+
+		azureClientSecret, err := getKmsCredential(azure, "clientSecret", envAzureClientSecret)
+		if err != nil {
+			return err
+		}
+		if azureClientSecret != nil {
+			kmsProviders["azure"]["clientSecret"] = *azureClientSecret
+		}
+	}
+
+	if gcp, ok := ceo.KmsProviders["gcp"]; ok {
+		kmsProviders["gcp"] = make(map[string]interface{})
+
+		gcpEmail, err := getKmsCredential(gcp, "email", envGcpEmail)
+		if err != nil {
+			return err
+		}
+		if gcpEmail != nil {
+			kmsProviders["gcp"]["email"] = *gcpEmail
+		}
+
+		gcpPrivateKey, err := getKmsCredential(gcp, "privateKey", envGcpPrivateKey)
+		if err != nil {
+			return err
+		}
+		if gcpPrivateKey != nil {
+			kmsProviders["gcp"]["privateKey"] = *gcpPrivateKey
+		}
+	}
+
+	if kmip, ok := ceo.KmsProviders["kmip"]; ok {
+		kmsProviders["kmip"] = make(map[string]interface{})
+
+		kmipEndpoint, err := getKmsCredential(kmip, "endpoint", "localhost:5698")
+		if err != nil {
+			return err
+		}
+		if kmipEndpoint != nil {
+			kmsProviders["kmip"]["endpoint"] = *kmipEndpoint
+		}
+	}
+
+	if local, ok := ceo.KmsProviders["local"]; ok {
+		kmsProviders["local"] = make(map[string]interface{})
+
+		defaultLocalKeyBase64 := "Mng0NCt4ZHVUYUJCa1kxNkVyNUR1QURhZ2h2UzR2d2RrZzh0cFBwM3R6NmdWMDFBMUN3YkQ5aXRRMkhGRGdQV09wOGVNYUMxT2k3NjZKelhaQmRCZGJkTXVyZG9uSjFk"
+		localKey, err := getKmsCredential(local, "key", defaultLocalKeyBase64)
+		if err != nil {
+			return err
+		}
+		if localKey != nil {
+			kmsProviders["local"]["key"] = *localKey
+		}
+	}
+
+	em.keyVaultClientIDs[ceo.KeyVaultClient] = true
+	keyVaultClient, ok := em.clientEntities[ceo.KeyVaultClient]
+	if !ok {
+		return newEntityNotFoundError("client", ceo.KeyVaultClient)
+	}
+
+	ce, err := mongo.NewClientEncryption(keyVaultClient.Client, options.ClientEncryption().SetKeyVaultNamespace(ceo.KeyVaultNamespace).SetKmsProviders(kmsProviders))
```

Comment:
Optional: Consider wrapping this line to make it easier to read.

E.g.
```go
ce, err := mongo.NewClientEncryption(
 	keyVaultClient.Client,
 	options.ClientEncryption().
 	 	SetKeyVaultNamespace(ceo.KeyVaultNamespace).
 	 	SetKmsProviders(kmsProviders))
```

Reply:
Done.